internal/mtail/unix_socket_export_integration_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+// This file is available under the Apache license.
+// +build integration
+
+package mtail_test
+
+import (
+	"fmt"
+	"path"
+	"testing"
+	"time"
+	"net"
+	"net/http"
+	"os"
+	"runtime"
+	"context"
+	"bytes"
+	"encoding/json"
+
+	"github.com/golang/glog"
+	"github.com/google/mtail/internal/mtail"
+	"github.com/google/mtail/internal/testutil"
+	"github.com/google/mtail/internal/metrics"
+	"github.com/google/mtail/internal/watcher"
+)
+
+// makeServer makes a new Server for use in tests, but does not start
+// the server.  It returns the server, or any errors the new server creates.
+func makeServer(tb testing.TB, pollInterval time.Duration, enableFsNotify bool, options ...func(*mtail.Server) error) (*mtail.Server, error) {
+	tb.Helper()
+	w, err := watcher.NewLogWatcher(pollInterval, enableFsNotify)
+	if err != nil {
+		tb.Fatal(err)
+	}
+
+	return mtail.New(metrics.NewStore(), w, options...)
+}
+
+
+// startUNIXSocketServer creates a new Server serving through a UNIX
+// socket and starts it running. It returns the server, and a cleanup function.
+func startUNIXSocketServer(tb testing.TB, pollInterval time.Duration, enableFsNotify bool, options ...func(*mtail.Server) error) (*mtail.Server, func()) {
```

Comment:
Took me a while to realise why you can't reuse `TestStartServer()` but you should be able to reuse `TestMakeServer()` instead of adding your `makeServer()` above.

We could refactor `TestStartServer()` to notice if there's a BindUnixSocket optoin and change how the `net.DialTimeout` calls to wait for the server to start, if you like.  If not, please add a comment to the docstring explaining why this one is different.

Reply:
I will modify the docstring because refactor `TestStartServer` means that the other tests need also to be modified. It's a good idea but perhaps it need to be done later.